### PR TITLE
Feat sllw

### PR DESCRIPTION
**Pull request by smml1996**

### CODE REVIEW

File: src/unicorn/builder.rs [hunk 1]
```diff
@@ -531,6 +531,18 @@ impl ModelBuilder {
         self.reg_flow_ite(itype.rd(), sra_node);
     }
 
+    fn model_sllw(&mut self, rtype: RType) {
+        let thirtytwo = self.new_const(32);
+
+        // shift value should only be a value less or equal than 2^32
+        let mut shifted_r2 = self.new_sll(self.reg_node(rtype.rs2()), thirtytwo.clone());
```
Comment (mstarzinger) [inline src/unicorn/builder.rs]: The RISC-V spec says ...
```
SLLW, SRLW, and SRAW are RV64I-only instructions that are analogously defined but operate
on 32-bit values and produce signed 32-bit results. The shift amount is given by rs2[4:0]
```
The way I interpret this we need to mask out more from `rs2` and only use that lowest five bits in `rs2` as a shift amount. What do you think?

Comment (mstarzinger) [reply src/unicorn/builder.rs]: You can look at how I implemented `sll` which does something similar and only uses the lowest six bits ...
```
    fn model_sll(&mut self, rtype: RType) {
        // Only the low 6 bits of rs2 are considered for the shift amount.
        let mask_node = self.new_const(0x3f); // TODO: Make this a global constant.
        let amount_node = self.new_and_word(self.reg_node(rtype.rs2()), mask_node);
        let sll_node = self.new_sll(self.reg_node(rtype.rs1()), amount_node);
        self.reg_flow_ite(rtype.rd(), sll_node);
    }
```

Comment (smml1996) [reply src/unicorn/builder.rs]: thanks! working on it!

File: src/unicorn/builder.rs [hunk 2]
```diff
@@ -531,6 +531,23 @@ impl ModelBuilder {
         self.reg_flow_ite(itype.rd(), sra_node);
     }
 
+    fn model_sllw(&mut self, rtype: RType) {
+        let thirtytwo = self.new_const(32);
+
+        // Only the low 4 bits of rs2 are considered for the shift amount.
+        let mask_node = self.new_const(0xf); // TODO: Make this a global constant.
```
Comment (mstarzinger) [inline src/unicorn/builder.rs]: Why only 4 bits? I think one has to use 5 bits to express the full range of valid shift amounts between 0 and 31. The syntax from the spec saying n[0:4] is inclusive and means bits [0,1,2,3,4] as far as I know. The mask should be `0x1f` instead I think.

**Review by mstarzinger — CHANGES_REQUESTED** [review]

**Review by mstarzinger — APPROVED** [review]
Looks good. Thanks!

**Pull request merged**